### ISSUE TITLE
Syntax mistake in redirect regex label breaks other backends

### ISSUE THREAD

**Issue by dac388** [containous/traefik#6431]
Bug

### What did you see?

Deploying a service with this label
"traefik.frontend.redirect.regex=^https?://(test\\.beta\\.redacted\\.org|redacted\\.com)/(.*)"

Causes the (preexisting) service to be unavailable (504 bad gateway).
Also, any other services deployed after this label is deployed will also begin to 504.
If traefik is redeployed or restarted after a service is deployed with this label ALL backends are not found (404).

I cannot post the correct syntax because slashes are being removed when I click update comment.

Everywhere there are two backslashes shoudl be changed to four backlashes because of escape characters in docker-compose.yml.  We have that deployed and working now.

### What did you expect to see?

There is a syntax error in that label.  Traefik does produce an error identifying the incorrect syntax. 

I **_do_** expect this could make the service with the label unavailable.  I **_do not_** expect this would break other backends

### Output of `traefik version`: (_What version of Traefik are you using?_)

Confirmed behavior with 1.7.20 and 1.7.21 (both alpine).

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

traefik.toml
```toml
InsecureSkipVerify = true
logLevel = "INFO"
defaultEntryPoints = ["https","http"]

[traefikLog]

[accessLog]
  [accessLog.filters]
  [accessLog.fields]
    defaultMode = "keep"
[entryPoints]
  [entryPoints.ping]
  address = ":8090"
  [entryPoints.http]
  address = ":80"
  compress = true
    [entryPoints.http.proxyProtocol]
      trustedIPs = ["redacted"]
    [entryPoints.http.forwardedHeaders]
      trustedIPs = ["redacted", "redacted"]
  [entryPoints.https]
  address = ":443"
  compress = true
    [entryPoints.https.tls]
	minVersion = "VersionTLS11"
      [[entryPoints.https.tls.certificates]]          
		  certFile = "redacted.cer"
		  keyFile = "redacted.key"
    [entryPoints.https.proxyProtocol]
      trustedIPs = ["redacted"]
    [entryPoints.https.forwardedHeaders]
      trustedIPs = ["redacted", "redacted"]

[ping]
entryPoint = "ping"

[docker]
endpoint = "unix:///var/run/docker.sock"
domain = "docker.localhost"
watch = true
swarmmode = true
```

Service was deployed with docker-composer.yml, to Docker Datacenter using swarm mode orchestrator.

These are the other labels used in docker-compose.yml

          - "traefik.frontend.whiteList.sourceRange=${TRAEFIK_WHITELIST:-redacted}"
          - "traefik.docker.network=is-2"
          - "traefik.port=8080"
          - "traefik.frontend.rule=Host:${TRAEFIK_HOST}"
          - "traefik.frontend.passHostHeader=true"
          - "traefik.backend.loadbalancer.stickiness=true"
          - "traefik.frontend.headers.SSLRedirect=true"

**Comment by ldez** [comment]
Hello,

the regular expression without escape is a valid regular expression even if it's not the regular expression that you are thinking to use.

So it's not possible to Traefik to detected this kind of syntax error.

Thanks anyway.

**Comment by dac388** [comment]
Traefik does detect and log the syntax error, and this brought down all our websites, literally millions of users worldwide.

**Comment by ldez** [comment]
I understand that this error must have caused you a lot of stress, but there is no point in making hidden threats.

We (the Traefik team) are humans like you and we are trying our best to respond to all issues on this free and open source product.

If you don't agree with our decision, give us more information and we will analyze it.

In your last message, you provided important information, so I will try to find if we can improve that.

**Comment by ldez** [comment]
Could you provide (in a file in a comment) the exact configuration you used to cause this error, thank you.

**Comment by dac388** [comment]
Hidden threats?  Have no idea what you're talking about.

**Comment by traefiker** [comment]
Closed by #6446.
